deepspeed/runtime/fp16/onebit/zoadam.py

```diff
@@ -0,0 +1,352 @@
+'''
+Copyright 2020 The Microsoft DeepSpeed Team
+'''
+import types
+import torch
+import importlib
+import numpy as np
+import time
+import torch.distributed as dist
+
+from deepspeed.utils.logging import logger
+
+
+class ZeroOneAdam(torch.optim.Optimizer):
+    """Implements the 0/1 Adam algorithm. Currently GPU-only.
+    For usage example please see https://www.deepspeed.ai/tutorials/zero-one-adam/
+    For technical details please read https://arxiv.org/abs/2202.06009
+    Arguments:
+        params (iterable): iterable of parameters to optimize or dicts defining
+            parameter groups.
+        lr (float, optional): learning rate. (default: 1e-3)
+        betas (Tuple[float, float], optional): coefficients used for computing
+            running averages of gradient and its square. (default: (0.9, 0.999))
+        eps (float, optional): term added to the denominator to improve
+            numerical stability. (default: 1e-8)
+        weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
+        var_freeze_step (int, optional): The latest step to update the variance,
+            using the notation from https://arxiv.org/abs/2202.06009, it denotes the
+            max{i|i in T_v}. Note that this is different from the freeze step from the
+            1-bit Adam. The var_freeze_step is usually the end of the learning rate warmup
+            and thus does not require tuning. (default: 100000)
+        var_update_scaler (int, optional): The interval to update the variance. Note that
+            the update policy for variance follows an exponential rule, where var_update_scaler
+            denotes the kappa in the 0/1 Adam paper. (default: 16)
+        local_step_scaler (int, optional): The interval to scale the local steps interval
+            according to the learning rate policy. (default: 32678)
+        local_step_clipper (int, optional): The largest interval for local steps with
+            learning rate policy. This corresponds to the variable H in the 0/1 Adam paper.
+            (default: 16)
```

Comment:
For these 4 hyperparameters for 0/1 Adam, when writing the tutorial please make sure to include a more detailed description (can refer the sections or equations in paper) and a tuning strategy.